Segmentation crash on image scanning

# Description
Kubescape started to crash on image scanning after upgrading to v4.x

# Environment
OS: `Darwin Kernel Version 25.5.0` <!-- the OS + version you’re running Kubescape on, e.g Ubuntu 22.04 LTS  -->
Version: ```
Your current version is: v4.0.9
Build commit: 002e791cd39fed51dd4a86b321c6d184fa672349
Build date: 2026-05-29T06:45:08Z
``` <!-- the version that Kubescape reports when you run `kubescape version`  -->
 
# Steps To Reproduce
<!--
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
-->

Scan any local image, with or without extra config flags:
```shell
.bin/kubescape scan image ghcr.io/k3d-io/k3d-proxy:5.9.0
 ✅  Successfully scanned image: ghcr.io/k3d-io/k3d-proxy:5.9.0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x170 pc=0x105e3a094]

goroutine 1 [running]:
github.com/kubescape/kubescape/v3/core/pkg/resultshandling.(*ResultsHandler).HandleResults(0x14002687a60, {0x108b98620, 0x140014d5220}, 0x14000f2bb08)
	/home/runner/work/kubescape/kubescape/core/pkg/resultshandling/results.go:80 +0x34
github.com/kubescape/kubescape/v3/core/core.(*Kubescape).ScanImage(0x14000cd4550, 0x14001530e10, 0x14000f2bb08)
	/home/runner/work/kubescape/kubescape/core/core/image_scan.go:232 +0x5b0
github.com/kubescape/kubescape/v3/cmd/scan.getImageCmd.func2(0x140014eec08, {0x14000e9c780, 0x1, 0x1065df0b9?})
	/home/runner/work/kubescape/kubescape/cmd/scan/image.go:67 +0x1c4
github.com/spf13/cobra.(*Command).execute(0x140014eec08, {0x14000e9c740, 0x1, 0x1})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1015 +0x7d4
github.com/spf13/cobra.(*Command).ExecuteC(0x14000f2b808)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1071
github.com/kubescape/kubescape/v3/cmd.Execute({0x108b98620, 0x140014d5220}, {0x1077c1018, 0x6}, {0x1077c2c40, 0x28}, {0x1077c2740, 0x14})
	/home/runner/work/kubescape/kubescape/cmd/root.go:123 +0xa4
main.main()
	/home/runner/work/kubescape/kubescape/main.go:45 +0x188

.bin/kubescape scan image ubuntu:latest --exceptions .github/kubescape-exceptions.json --controls-config .github/kubescape-control.json --view control --keep-local
 ✅  Successfully scanned image: ubuntu:latest
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x170 pc=0x106ab6094]

goroutine 1 [running]:
github.com/kubescape/kubescape/v3/core/pkg/resultshandling.(*ResultsHandler).HandleResults(0x14004687a60, {0x109814620, 0x14000c6a780}, 0x14000ebcf08)
	/home/runner/work/kubescape/kubescape/core/pkg/resultshandling/results.go:80 +0x34
github.com/kubescape/kubescape/v3/core/core.(*Kubescape).ScanImage(0x14000f8c9a0, 0x140014e4370, 0x14000ebcf08)
	/home/runner/work/kubescape/kubescape/core/core/image_scan.go:232 +0x5b0
github.com/kubescape/kubescape/v3/cmd/scan.getImageCmd.func2(0x14001486008, {0x14000e9e700, 0x1, 0x10725b0b9?})
	/home/runner/work/kubescape/kubescape/cmd/scan/image.go:67 +0x1c4
github.com/spf13/cobra.(*Command).execute(0x14001486008, {0x14000e9e680, 0x8, 0x8})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1015 +0x7d4
github.com/spf13/cobra.(*Command).ExecuteC(0x14000ebcc08)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1148 +0x350
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.10.2/command.go:1071
github.com/kubescape/kubescape/v3/cmd.Execute({0x109814620, 0x14000c6a780}, {0x10843d018, 0x6}, {0x10843ec40, 0x28}, {0x10843e740, 0x14})
	/home/runner/work/kubescape/kubescape/cmd/root.go:123 +0xa4
main.main()
	/home/runner/work/kubescape/kubescape/main.go:45 +0x188
```

# Expected behavior
<!-- A clear and concise description of what you expected to happen. -->
Image scanner works as in release 3.x

# Actual Behavior
<!-- A clear and concise description of what happened. If applicable, add screenshots to help explain your  problem. -->

# Additional context
<!-- Add any other context about the problem here. -->

--- a/core/pkg/resultshandling/results.go
+++ b/core/pkg/resultshandling/results.go
@@ -77,7 +77,7 @@ func (rh *ResultsHandler) GetResults() *reporthandlingv2.PostureReport {
 
 // HandleResults handles all necessary actions for the scan results
 func (rh *ResultsHandler) HandleResults(ctx context.Context, scanInfo *cautils.ScanInfo) error {
-	if len(rh.ScanData.VAPPolicies) > 0 {
+	if rh.ScanData != nil && len(rh.ScanData.VAPPolicies) > 0 {
 		index := vapreconcile.BuildIndex(rh.ScanData.VAPPolicies, rh.ScanData.VAPBindings)
 		vapreconcile.EnrichSummary(rh.ScanData.Report.SummaryDetails.Controls, index)
 	}

--- a/core/pkg/resultshandling/results_test.go
+++ b/core/pkg/resultshandling/results_test.go
@@ -61,6 +61,33 @@ func TestResultsHandlerHandleResultsPrintsResultsToUI(t *testing.T) {
 	}
 }
 
+// TestResultsHandlerHandleResultsImageScanNilScanData reproduces issue #2430:
+// image scans construct the handler with a nil ScanData (only ImageScanData is
+// set), so HandleResults must not dereference ScanData. Before the fix this
+// panicked with a nil pointer dereference in the VAP-reconcile block.
+func TestResultsHandlerHandleResultsImageScanNilScanData(t *testing.T) {
+	uiPrinter := &SpyPrinter{}
+	outputPrinter := &SpyPrinter{}
+
+	// Mirror core.ScanImage: nil reporter, nil ScanData, only ImageScanData.
+	rh := NewResultsHandler(nil, []printer.IPrinter{outputPrinter}, uiPrinter)
+	rh.ImageScanData = []cautils.ImageScanData{{}}
+
+	assert.Nil(t, rh.ScanData)
+
+	var err error
+	assert.NotPanics(t, func() {
+		err = rh.HandleResults(context.Background(), &cautils.ScanInfo{})
+	})
+	assert.NoError(t, err)
+
+	// Results are still printed even though ScanData is nil...
+	assert.Equal(t, 1, uiPrinter.ActionPrintCalls)
+	assert.Equal(t, 1, outputPrinter.ActionPrintCalls)
+	// ...but the compliance score is skipped, as it requires ScanData.
+	assert.Equal(t, 0, outputPrinter.ScoreCalls)
+}
+
 func TestValidatePrinter(t *testing.T) {
 	tests := []struct {
 		name        string

